File: src/types/review-template.ts
// Types for Review Templates

export interface ReviewTemplate {
  id: number;
  name: string;
  icon: string;
  color: string;
  content: string;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateReviewTemplateDto {
  name: string;
  icon: string;
  color?: string;
  content: string;
}

export interface UpdateReviewTemplateDto {
  template_id: number;
  name?: string;
  icon?: string;
  color?: string;
  content?: string;
}

File: src-tauri/src/models/review_template.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewTemplate {
    pub id: i64,
    pub name: String,
    pub icon: String,
    pub color: String,
    pub content: String,
    pub is_default: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateReviewTemplateDto {
    pub name: String,
    pub icon: String,
    pub color: Option<String>,
    pub content: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateReviewTemplateDto {
    pub template_id: i64,
    pub name: Option<String>,
    pub icon: Option<String>,
    pub color: Option<String>,
    pub content: Option<String>,
}
